File: Successful-Pairs-of-Spells-and-Potions.cs
public class Solution {
    public int[] SuccessfulPairs(int[] spells, int[] potions, long success){
        int n = spells.Length;
        int m = potions.Length;
        int[] res = new int[n];
        Array.Fill(res,0);
        Array.Sort(potions);

        for(int i = 0;i<n;i++){
            int l= 0,r = m ;
            while(l<r){
               int mid = l + (r-l)/2 ; 
               if((long)spells[i] * potions[mid]<success){
                    l = mid+1;
               }

               else if((long)spells[i] * potions[mid]>=success){
                    r = mid;
               }

               }

               res[i] = m-l;

            }
        return res;
    }
}

File: Successful-Pairs-of-Spells-and-Potions.cpp
class Solution {
public:
    vector<int> successfulPairs(vector<int>& spells, vector<int>& potions, long long success) {
        vector<int> res(spells.size(),0);
        sort(potions.begin(),potions.end());
        for(int i = 0 ;i<spells.size();i++){
            int l = 0 , r = potions.size();
            while(l<r){
                int m = l + (r-l)/2 ;
                if((long long)spells[i]*potions[m]>=success){
                    r = m ;
                }else{
                    l = m+1 ; 
                }
            }

            res[i] = potions.size()-l ; 
        }

        return res;

    }
};